core/crypto/crypto.c

```diff
@@ -162,43 +162,107 @@ TEE_Result crypto_cipher_get_block_size(uint32_t algo, size_t *size)
 	}
 }
 
-#if !defined(_CFG_CRYPTO_WITH_MAC)
-TEE_Result crypto_mac_alloc_ctx(void **ctx __unused, uint32_t algo __unused)
+TEE_Result crypto_mac_alloc_ctx(void **ctx, uint32_t algo)
 {
-	return TEE_ERROR_NOT_IMPLEMENTED;
+	TEE_Result res __maybe_unused = TEE_SUCCESS;
+	struct crypto_mac_ctx *c __maybe_unused = NULL;
+
+	switch (algo) {
+#if defined(CFG_CRYPTO_HMAC)
```

Comment:
Same suggestion here about moving conditionals to `crypto_impl.h`.

Reply:
Yes, I'll take care of all those.